http/src/main/resources/jupyter/gce-init.sh

```diff
@@ -0,0 +1,528 @@
+SERVICE_ACCOUNT_CREDENTIALS=""
+if [ ! -z ${SERVICE_ACCOUNT_CREDENTIALS} ] ; then
+  gsutil cp ${SERVICE_ACCOUNT_CREDENTIALS} /etc
+  SERVICE_ACCOUNT_CREDENTIALS=`basename ${SERVICE_ACCOUNT_CREDENTIALS}`
+  export GOOGLE_APPLICATION_CREDENTIALS=/etc/${SERVICE_ACCOUNT_CREDENTIALS}
+fi
+
+#####################################################################################################
+# Set up that IS specific to GCE_OPERATION:
+#
+# We perform some of the operations based on whether we are 'creating' a GCE VM or 'restarting'
+# a VM that was previously created and stopped.
+#####################################################################################################
+
+if [[ "${GCE_OPERATION}" == 'creating' ]]; then
+    JUPYTER_SCRIPTS=${JUPYTER_HOME}/scripts
+    JUPYTER_USER_HOME=/home/jupyter-user
+    KERNELSPEC_HOME=/usr/local/share/jupyter/kernels
+
+    # The following values are populated by Leo when a cluster is created.
+    export RSTUDIO_SERVER_NAME=$(rstudioServerName)
+    export PROXY_SERVER_NAME=$(proxyServerName)
+    export RSTUDIO_DOCKER_IMAGE=$(rstudioDockerImage)
+    export PROXY_DOCKER_IMAGE=$(proxyDockerImage)
+    export MEM_LIMIT=$(memLimit)
+
+    SERVER_CRT=$(jupyterServerCrt)
+    SERVER_KEY=$(jupyterServerKey)
+    ROOT_CA=$(rootCaPem)
+    JUPYTER_DOCKER_COMPOSE=$(jupyterDockerCompose)
+    RSTUDIO_DOCKER_COMPOSE=$(rstudioDockerCompose)
+    PROXY_DOCKER_COMPOSE=$(proxyDockerCompose)
+    WELDER_DOCKER_COMPOSE=$(welderDockerCompose)
+    PROXY_SITE_CONF=$(proxySiteConf)
+    JUPYTER_SERVER_EXTENSIONS=$(jupyterServerExtensions)
+    JUPYTER_NB_EXTENSIONS=$(jupyterNbExtensions)
+    JUPYTER_COMBINED_EXTENSIONS=$(jupyterCombinedExtensions)
+    JUPYTER_LAB_EXTENSIONS=$(jupyterLabExtensions)
+    JUPYTER_USER_SCRIPT_URI=$(jupyterUserScriptUri)
+    JUPYTER_USER_SCRIPT_OUTPUT_URI=$(jupyterUserScriptOutputUri)
+    JUPYTER_NOTEBOOK_CONFIG_URI=$(jupyterNotebookConfigUri)
+    JUPYTER_NOTEBOOK_FRONTEND_CONFIG_URI=$(jupyterNotebookFrontendConfigUri)
+    CUSTOM_ENV_VARS_CONFIG_URI=$(customEnvVarsConfigUri)
+
+    STEP_TIMINGS+=($(date +%s))
+
+    log 'Copying secrets from GCS...'
+
+    mkdir /work
+    mkdir /certs
+    chmod a+rwx /work
+
+    # Add the certificates from the bucket to the VM. They are used by the docker-compose file
+    gsutil cp ${SERVER_CRT} /certs
+    gsutil cp ${SERVER_KEY} /certs
+    gsutil cp ${ROOT_CA} /certs
+    gsutil cp ${PROXY_SITE_CONF} /etc
+    gsutil cp ${JUPYTER_DOCKER_COMPOSE} /etc
+    gsutil cp ${RSTUDIO_DOCKER_COMPOSE} /etc
+    gsutil cp ${PROXY_DOCKER_COMPOSE} /etc
+    gsutil cp ${WELDER_DOCKER_COMPOSE} /etc
+
+    # Needed because docker-compose can't handle symlinks
+    touch /hadoop_gcs_connector_metadata_cache
+
+    # Not all images have the directory used for Stackdriver configs. If so, create it
+    mkdir -p /etc/google-fluentd/config.d
+
+    # Add stack driver configuration for welder
+    tee /etc/google-fluentd/config.d/welder.conf << END
```

Comment:
can these 3 stackdriver configurations be bundled into a function? It seems like the format and 'tool' are the only parameters needed

Reply:
+1

Reply:
Good idea. These will probably move to the GCE image, though. Will keep in mind during that work.